sentry_sdk/integrations/chalice.py

```diff
@@ -0,0 +1,182 @@
+from sentry_sdk.hub import Hub, _should_send_default_pii
+from sentry_sdk.integrations import Integration
+from sentry_sdk.integrations._wsgi_common import _filter_headers
+from sentry_sdk.integrations.aws_lambda import AwsLambdaIntegration
+from sentry_sdk.utils import (
+    AnnotatedValue,
+    capture_internal_exceptions,
+    event_from_exception,
+)
+
+if MYPY:
+    from typing import Any, Optional
+
+    from sentry_sdk._types import Event, EventProcessor, Hint
+
+
+def _get_view_function_response(app, view_function, function_args):
+    hub = Hub.current
+    client = hub.client
+
+    with hub.push_scope() as scope:
+        try:
+            response = view_function(**function_args)
+            if not isinstance(response, Response):
+                response = Response(body=response)
+            app._validate_response(response)
+        except ChaliceViewError as e:
+            # Any chalice view error should propagate.  These
+            # get mapped to various HTTP status codes in API Gateway.
+            response = Response(
+                body={'Code': e.__class__.__name__, 'Message': str(e)},
+                status_code=e.STATUS_CODE,
+            )
+            hub.flush()
+        except Exception:
+            with capture_internal_exceptions():
+                scope.transaction = app.lambda_context.function_name
+                scope.add_event_processor(
+                    _make_request_event_processor(
+                        app.current_request, app.lambda_context
+                    )
+                )
+            exc_info = sys.exc_info()
+            event, hint = event_from_exception(
+                exc_info,
+                client_options=client.options,
+                mechanism={"type": "chalice", "handled": False},
+            )
+            hub.capture_event(event, hint=hint)
+            hub.flush()
+            headers = {}
+            app.log.error(
+                "Caught exception for %s", view_function, exc_info=True
+            )
+            if app.debug:
+                # If the user has turned on debug mode,
+                # we'll let the original exception propagate so
+                # they get more information about what went wrong.
+                stack_trace = ''.join(traceback.format_exc())
+                body = stack_trace
+                headers['Content-Type'] = 'text/plain'
+            else:
+                body = {
+                    'Code': 'InternalServerError',
+                    'Message': 'An internal server error occurred.',
+                }
+            response = Response(body=body, headers=headers, status_code=500)
+    return response
+
+
+class ChaliceIntegration(Integration):
+    identifier = "chalice"
+
+    @staticmethod
+    def setup_once():
+        # for @app.route()
+        Chalice._get_view_function_response = _get_view_function_response
+        # for everything else (like events)
+        AwsLambdaIntegration.setup_once()
```

Comment:
Would just document that one needs to enable this. Unfortunately we have no way to enforce this programmatically right now. What you're doing here will enable the integration twice, meaning double the instrumentation overhead.

Reply:
i actually found a better way to catch exceptions in events running in chalice, which doesn't require adding the aws lambda integration https://github.com/getsentry/sentry-python/blob/f1443129021cd1d4a61425b293eda49ec712200f/sentry_sdk/integrations/chalice.py#L113